File: patterns/rule-manager/rules-python/rule_manager_toggle.py
from openhab import rule, Registry
from openhab.services import getService
from openhab.triggers import ItemStateChangeTrigger

rule_manager = getService("org.openhab.core.automation.RuleManager")


@rule(triggers=[ItemStateChangeTrigger("exampleRule")])
class EnableDisableExampleRule:
    def execute(self, module, input):
        enabled = str(Registry.getItem("exampleRule").getState()) == "ON"
        rule_manager.setEnabled("example_rule_uid", enabled)
        self.logger.info("example_rule_uid enabled=" + str(enabled))

File: patterns/rule-manager/rules-js/rule-manager-toggle.js
rules.JSRule({
  name: "Enable/Disable example rule",
  triggers: [triggers.ItemStateChangeTrigger("exampleRule")],
  execute: (event) => {
    const enable = items.getItem("exampleRule").state === "ON";
    try {
      rules.setEnabled("example_rule_uid", enable);
      console.log("example_rule_uid -> enabled=" + enable);
    } catch (e) {
      console.warn("Konnte example_rule_uid nicht umschalten: " + e.message);
    }
  }
});
